init.lua

```diff
@@ -61,14 +65,45 @@ function filter.register_on_violation(func)
 	table.insert(filter.registered_on_violations, func)
 end
 
-function filter.check_message(name, message)
+-- Filters return a warning string if triggered
+function filter.register_filter(name, func)
+	filter.registered_filters[name] = func
+end
+
+--
+-- Filters
+--
+
+-- Bad words filter
+filter.register_filter("bad_words", function(name, message)
 	for _, w in ipairs(words) do
 		if string.find(message:lower(), "%f[%a]" .. w .. "%f[%A]") then
-			return false
+			return "Watch your language!"
+		end
+	end
+end)
+
+-- check_message invokes all filters, and returns true and a '\n'
+-- separated list of warnings if at least one filter is triggered
+function filter.check_message(name, message)
+	local str
+	if not ret_val[name] then
+		ret_val[name] = {}
+	end
+
+	for fname, component in pairs(filter.registered_filters) do
```

Comment:
`component` is very confusing. It's a `func` or `fn`.

Reply:
if `fname` is unused, do

```
for _, fn in ....
```

Reply:
Couldn't think of a better alternative to `component`, `fn` is way better...